test/onnx/torch_export/test_torch_export_with_onnxruntime.py

```diff
@@ -0,0 +1,298 @@
+            input_b = torch.randn(3, 3)
+            model = CustomModule()
+
+            dim = torch.export.Dim("dim")
+            exported_program = torch.export.export(
+                model,
+                (
+                    input_x,
+                    input_b,
+                ),
+                dynamic_shapes=({0: dim}, {0: dim}),
+            )
+            onnx_program = torch.onnx.dynamo_export(exported_program, input_x, input_b)
+
+            # different dim inputs
+            additional_inputs_x = torch.rand((4, 3), dtype=torch.float32)
+            additional_inputs_b = torch.randn(4, 3)
+            self._compare_onnx_and_torch_exported_program(
+                exported_program,
+                onnx_program,
+                (
+                    additional_inputs_x,
+                    additional_inputs_b,
+                ),
+            )
+
+    def test_onnx_program_supports_non_arg_name_with_container_type(self):
+        def foo(a, b):
+            return a[0].sum() + a[1].sum() + b.sum()
+
+        inp_a = (torch.randn(4, 4), torch.randn(4, 4))
+        inp_b = torch.randn(4, 4)
+        inp = (inp_a, inp_b)
+
+        count = 0
+
+        def dynamify_inp(x):
+            # Mark the second input a[1] dynamic
+            nonlocal count
+            if count == 1:
+                dim = torch.export.Dim("dim", min=3)
+                count += 1
+                return {0: dim}
+            count += 1
+            return None
+
+        dynamic_shapes = torch_pytree.tree_map(dynamify_inp, inp)
+        exported_program = torch.export.export(foo, inp, dynamic_shapes=dynamic_shapes)
+        onnx_program = torch.onnx.dynamo_export(exported_program, inp_a, inp_b)
+
+        # NOTE: Careful with the input format. The input format should be
+        # consistent with how the model is exported.
+        test_inputs = ((torch.randn(4, 4), torch.randn(6, 4)), torch.randn(4, 4))
+        self._compare_onnx_and_torch_exported_program(
+            exported_program, onnx_program, test_inputs
+        )
+
+    def test_onnx_program_supports_lazy_module_kwargs(self):
+        class LazyModule(torch.nn.modules.lazy.LazyModuleMixin, torch.nn.Module):
+            def initialize_parameters(self, *args, **kwargs):
+                pass
+
+            def forward(self, x, y):
+                return x + y
+
+        m = LazyModule()
+        dim = torch.export.Dim("dim")
+        dynamic_shapes = ({0: dim}, {0: dim})
+        exported_program = torch.export.export(
+            m,
+            (),
+            {"x": torch.randn(3, 3), "y": torch.randn(3, 3)},
+            dynamic_shapes=dynamic_shapes,
+        )
+        onnx_program = torch.onnx.dynamo_export(
+            exported_program, x=torch.randn(3, 3), y=torch.randn(3, 3)
+        )
+
+        # NOTE: A model should be fed with the input formats that
+        # how the model is exported
+        inputs = {"x": torch.randn(6, 3), "y": torch.randn(6, 3)}
```

Comment:
same as above. use `onnx_program.adapt_torch_inputs_tyo_onnx()`

Reply:
The note is kept for torch exported program usage.